common_utils/views_stepshifter_darts/hurdle_model.py

```diff
@@ -0,0 +1,66 @@
+from .stepshifter import StepshifterModel
+from .validation import views_validate
+from views_forecasts.extensions import *
+from sklearn.utils.validation import check_is_fitted
+
+
+class HurdleModel(StepshifterModel):
+    """
+    Hurdle model for time series forecasting. The model consists of two stages:
+    1. Binary stage: Predicts whether the target variable is 0 or > 0.
+    2. Positive stage: Predicts the value of the target variable when it is > 0.
+
+    ! Note:
+    We use a lightly different algorithm here. For the binary stage, a regression model is used, but the target is binary (0 or 1).
+    For the positive stage, a regression model is used as well. We select countries/priogrids that have at least one positive value in the target variable.
+    Since the target variable can be a float number, we use a threshold (default 0.1) to determine whether the target variable is positive or not.
+    """
+    def __init__(self, config: Dict, partitioner_dict: Dict[str, List[int]]):
+        super().__init__(config, partitioner_dict)
+        self._clf = self._resolve_estimator(config['model_clf'])
+        self._reg = self._resolve_estimator(config['model_reg'])
+        self._clf_params = self._get_parameters(config)['clf']
+        self._reg_params = self._get_parameters(config)['reg']
+
+    @views_validate
+    def fit(self, df: pd.DataFrame, threshold: float = 0.1):
+        self._prepare_time_series(df)
+
+        # Binary outcome (event/no-event)
+        # According to the DARTS doc, if timeseries uses a numeric type different from np.float32 or np.float64, not all functionalities may work properly.
+        # So use astype(float) instead of astype(int) (we should have binary outputs 0,1 though)
+        target_binary = [s.map(lambda x: (x > threshold).astype(float)) for s in self._target_train]
+
+        # Positive outcome (for cases where target > 0)
```

Comment:
I get your point but I think this would be more clear:

Positive outcome (for cases where target > threshold)